### PR TITLE
Enhancement/focus

### PR DESCRIPTION
**Pull request by niekvanderlaan**
With this pull request, nodes that were previously collapsed under the focused nodes will be highlighted.

### CODE REVIEW

File: src/main/java/application/fxobjects/cell/graph/BubbleCell.java [hunk 1]
```diff
@@ -75,6 +75,21 @@ public void focus() {
     }
 
     /**
+     * Method to set the side focus.
+     */
+    public void sideFocus() {
```
Comment (ghost) [inline src/main/java/application/fxobjects/cell/graph/BubbleCell.java]: Can't this code be implemented in an interface where each Cell inherits the sideFocus behaviour? As copy-pasting code in multiple classes is not very good for maintainibility.

See also: http://howtodoinjava.com/object-oriented/multiple-inheritance-in-java/

File: src/main/java/core/Node.java [hunk 2]
```diff
@@ -470,4 +476,38 @@ public int getNucleotides() {
...
+     * @return PreviousLevelNodesIds
+     */
+    public ArrayList<Integer> getPreviousLevelNodesIds() {
+        return previousLevelNodesIds;
+    }
+
+    /**
+     * Method to set the previousLevelNodesIds
+     * @param previousLevelNodesIds the list of Ids of all nodes collapsed
+     *                              in the previous level.
```
Comment (ghost) [inline src/main/java/core/Node.java:491]: Comment should continue at the beginning of a new line

File: src/main/java/core/Node.java [hunk 3]
```diff
@@ -470,4 +476,38 @@ public int getNucleotides() {
...
+     * @param previousLevelNodeId the nodeId to be added to the list.
+     */
+    public void addPreviousLevelNodesId(int previousLevelNodeId) {
+        this.previousLevelNodesIds.add(previousLevelNodeId);
+    }
+
+    /**
+     * Adds a list of nodes to the previousLevelNodesIds
+     * @param previousLevelNodesIds the list of Ids of all nodes collapsed
+     *                              in the previous level.
```
Comment (ghost) [inline src/main/java/core/Node.java:508]: Comment should continue at the beginning of a new line

**Comment by niekvanderlaan** [comment]
Thanks for the addition! I reviewed the code and it looks good. I will merge it into the master.

**Pull request merged**